ClientApp/viking-vault/src/Router.tsx

```diff
@@ -44,8 +45,31 @@ class Router extends React.Component<any, IState> {
     private makeRedirect() {
         if(sessionStorage.getItem("Authentication-Token") === null)
             return <Redirect to="/login" />;
-        else
-            return <Redirect to={this.state.isAdmin? "/admin" : "/user"} />;
+        else {
+            if(this.state.isAdmin === null) {
+                setTimeout(() =>{
+                    return <Redirect to={this.state.isAdmin? "/admin" : "/user"} />;
+                }, 100);
```

Comment:
Is this 100ms delay even noticeable?

Reply:
The delay is there because in that case we have to wait more for the response to arrive from backend.

Reply:
And what happens if the response takes more than 100ms?

Here's a better idea (though there are also surely other better approaches)

set the isAdmin to be false by default, and depending on the boolean state of that variable, render different `<Route>` components in `render()` method